megamek/src/megamek/client/ui/swing/BoardEditor.java

```diff
@@ -1440,12 +1440,32 @@ public String getDescription() {
         mapSettings.setBoardSize(board.getWidth(), board.getHeight());
         refreshTerrainList();
     }
+    
+    /**
+     * Will do board.initializeHex() for all hexes, correcting 
+     * building and road connection issues for those hexes that do not have
+     * the exits check set.
+     */
+    private void correctExits() {
+        for (int x = 0; x < board.getWidth(); x++) {
+            for (int y = 0; y < board.getHeight(); y++) {
+                board.initializeHex(x, y);
+            }
+        }
+    }
 
     /**
      * Checks to see if there is already a path and name stored; if not, calls
      * "save as"; otherwise, saves the board to the specified file.
      */
     private void boardSave() {
+        // First, correct connection issues and do a validation.
+        correctExits();
+        StringBuffer errBuff = new StringBuffer();
+        board.isValid(errBuff);
+        if (errBuff.length() > 0) {
+            showBoardValidationReport(errBuff);
+        }
```

Comment:
Why use a StringBuffer? Does this require thread safety?

Reply:
I didn't introduce it and I didn't really think about that.